### PR TITLE
JNG-5710 validation for identifiers cannot be decimal numbers

### PR DESCRIPTION
**Pull request by gaborflorian**
<table>
<td>
  <a href="https://blackbelt.atlassian.net/browse/JNG-5710" title="JNG-5710" target="_blank">TaskJNG-5710</a>  ESM and JSL validation rules for identifiers that cannot be decimal numbers
  </td></table>
  <br />
 

JNG-5710_validation_for_identifiers_cannot_be_decimal_numbers

### CODE REVIEW

**Review by robertcsakany — APPROVED** [review]

File: model/src/main/java/hu/blackbelt/judo/meta/jsl/validation/JslDslValidator.xtend [hunk 1]
```diff
@@ -1736,7 +1736,16 @@ class JslDslValidator extends AbstractJslDslValidator {
                 JsldslPackage::eINSTANCE.entityMemberDeclaration.getEStructuralFeature("ID"),
                 INVALID_DECLARATION)
     	}
-    	
+
+        if (field.identifier &&
+            field.referenceType instanceof PrimitiveDeclaration &&
+            TypeInfo.getTargetType(field.referenceType).isNumeric &&
+            ((field.referenceType as Modifiable).getModifier(JsldslPackage::eINSTANCE.scaleModifier) as ScaleModifier).value > BigInteger.ZERO ) {
+            error("An identifier field cannot be decimal number.",
```
Comment (robertcsakany) [inline model/src/main/java/hu/blackbelt/judo/meta/jsl/validation/JslDslValidator.xtend]: Change message to: Identifier cannot be numeric type with scale

**Review by robertcsakany — CHANGES_REQUESTED** [review]

File: model/src/main/java/hu/blackbelt/judo/meta/jsl/validation/JslDslValidator.xtend (hunk 1 above)
Comment (bencegelei) [inline model/src/main/java/hu/blackbelt/judo/meta/jsl/validation/JslDslValidator.xtend]: I suggest the following error message: "Identifier fields' type cannot be numeric with non-zero scale"

**Review by bencegelei — APPROVED** [review]

**Review by bencegelei — CHANGES_REQUESTED** [review]
accidental approve

File: model/src/main/java/hu/blackbelt/judo/meta/jsl/validation/JslDslValidator.xtend (hunk 1 above)
Comment (bencegelei) [reply model/src/main/java/hu/blackbelt/judo/meta/jsl/validation/JslDslValidator.xtend]: +1

Comment (gaborflorian) [reply model/src/main/java/hu/blackbelt/judo/meta/jsl/validation/JslDslValidator.xtend]: [c4cd2a4](https://github.com/BlackBeltTechnology/judo-meta-jsl/pull/170/commits/c4cd2a47652917be76d56b6678760cbf72164d63)

Comment (gaborflorian) [reply model/src/main/java/hu/blackbelt/judo/meta/jsl/validation/JslDslValidator.xtend]: [c4cd2a4](https://github.com/BlackBeltTechnology/judo-meta-jsl/pull/170/commits/c4cd2a47652917be76d56b6678760cbf72164d63)

**Review by noherczeg — APPROVED** [review]

**Review by robertcsakany — APPROVED** [review]

**Pull request merged**